Sessions hiding on fake passcode

moszatak1 opened the pull request:
TODO
 - [x] fix counter update on "Devices" ListItem
 - [x] check for better approach then `updated` closure param on `func hiddenSessionsController`

G-200 reviewed (commented):
In PasscodeOptionsController.swift, as I understand it, changeFakePasscode now means opening the options of the given passcode, so it's probably more logical to call it, for example, openFakePasscodeOptions.

Checked the rest, so far there are no questions.

G-200 reviewed (approved):
OK

moszatak1 committed:
Sessions hiding on fake passcode

moszatak1 committed:
Fix feedback + session counter update

moszatak1 committed:
Update sessions to hide feature

 - Simplify SessionSelectionMode enum
 - Remove unused code from ItemListSessionItem
 - Fix sessions to hide label on FakePasscodeAccountActionsController

moszatak1 committed:
Add missing dataPromise weak refs
